python/tk_multi_workfiles/entity_models/deferred_model.py

```diff
@@ -519,6 +520,23 @@ def _on_deferred_data_refreshed(self, sg_entity, changed, pending_refresh=False)
                 # The item might be already gone because one its ancestor was
                 # deleted.
                 self._delete_item(item)
+        # If we have any uid which is not in both sets, then things changed.
+        if existing_uids ^ refreshed_uids:
+            # Make sure we only trigger a refresh only once after all pending
+            # deferred data has been processed by posting a single data_refreshed
+            # emission in the event queue after all events which are already queued.
+            if not self._pending_delayed_data_refreshed:
```

Comment:
this feels a bit entangled. Do you think there would be a way to encapsulate it into an object for clarity? That object would hold the `_pending_delayed_data_refreshed` state internally and all we need to do was to emit the signal and implement the slot and the logic for making sure that it only fired once was kept within the object. Just something to consider to reduce the complexity of the runtime state of this class.

Reply:
I quickly tried the suggested object approach, but this makes things slightly more complicated, and in the end we would still need a member to store the object (if I understood well). So I added a method to encapsulate this code. We now simply have `self._post_delayed_data_refreshed()`, which, hopefully, makes the code clear enough and not so entangled.